seqr/views/apis/data_manager_api.py

```diff
@@ -395,6 +396,69 @@ def load_rna_seq_sample_data(request, sample_guid):
     return create_json_response({'success': True})
 
 
+def _load_phenotype_prioritization(file_path, user):
+    data_by_id = load_phenotype_prioritization_data_file(file_path)
+
+    all_samples = [sample for project_samples in data_by_id.values() for sample in project_samples.values()]
+    all_records = [rec for sample_records in all_samples for rec in sample_records]
+    message = f'Parsed {len(all_records)} LIRICAL/Exomiser data records in {len(all_samples)} samples'
+    info = [message]
+    logger.info(message, user)
+
+    for project, project_samples in data_by_id.items():
+        indivs = Individual.objects.filter(family__project__name=project, individual_id__in=project_samples.keys())
+        existing_indivs_by_id = {ind.individual_id: ind for ind in indivs}
+
+        tool_sample_id_set = set()
+        for sample_id, records in project_samples.items():
+            if existing_indivs_by_id[sample_id]:
+                for rec in records:
+                    rec['individual'] = existing_indivs_by_id[sample_id]
+                    tool_sample_id_set.add(f'{rec["tool"]}{sample_id}')
+            else:
+                raise ValueError(f'Individual {sample_id} doesn\'t exist in project {project}')
+
+        # Delete old data
+        to_delete = PhenotypePrioritization.objects.annotate(
+            tool_ind=Concat('tool', 'individual__individual_id', output_field=TextField())
+        ).filter(
+            tool_ind__in=tool_sample_id_set,
+        )
+        if to_delete:
+            deleted, _ = PhenotypePrioritization.bulk_delete(user, to_delete, parent='individual')
+            message = f'Deleted {deleted} existing phenotype-based prioritization records from project {project}'
```

Comment:
we should use the tool names here instead of "phenotype-based prioritization"

Reply:
Good point.